lib/entities/quote/DutchQuote.ts

```diff
@@ -426,8 +426,12 @@ export class DutchQuote implements IQuote {
     return this.portionAmountOutStart.mul(this.amountInStart).div(this.amountOutStart.add(this.portionAmountOutStart));
   }
 
+  public get amountInGasAdjusted(): BigNumber {
+    return this.amountIn.sub(this.portionAmountInStart);
+  }
+
   public get amountInGasAndPortionAdjusted(): BigNumber {
-    return this.amountIn.add(this.portionAmountInStart);
+    return this.amountIn;
```

Comment:
I'm a bit confused, previously this was returning
`amountIn + portionIn`

now, it's returning 
`amountIn (prev) - portionIn`

Isn't this double-adjusting the portion now?

Reply:
also, now this function naming is wrong, no? That seems confusing for the future - would it be more clear to just change the comparison point [here](https://github.com/Uniswap/unified-routing-api/blob/6b5f4fa18ecac61ff5a6f3d8e23a02eb349d1a77/lib/handlers/quote/handler.ts#L470) to use gasAdjusted?

Reply:
I think we want to still use `amountInGasAndPortionAdjusted`. The reason is because the amount here would be the exact out quote amount without trying to subtract the implied exchange rate of the interface fee at output token, so it should be more accurate.

Good catch on the double adjustment.. I think the best solution here is to for `get amountIn()` to stay the same, because so many places in the code touches it. Now `amountInGasAndPortionAdjusted` just need to remove `.add(this.portionAmountInStart)`